nixos/modules/services/monitoring/grafana.nix

```diff
@@ -78,6 +84,64 @@ in {
       type = types.str;
     };
 
+    datasourceConfigText = mkOption {
+      type = types.nullOr types.lines;
+      default = null;
+      description = ''
+        Datasource configuration as YAML text. If non-null, this option
+        defines the text that is written to datasource_conf.yml.
+        Avoid two apostrophes in a row inside YAML as empty value
+        or replace to double quote, because it interfere with Nix syntax!
+        Example:
+
+        datasources:
+          - name: Prometheus
+            type: prometheus
+            access: proxy
+            orgId: 1
+            url: http://localhost:9090
+            isDefault:
+            jsonData:
+              graphiteVersion: '1.1'
+            version: 1
+            editable: false
+      '';
+    };
+
+    dashboardPastedJson = mkOption {
+      type = types.nullOr types.lines;
+      default = null;
+      description = ''
+        Dashboard configuration as JSON text. If non-null, this option
+        defines the text that is written to dashboard.json.
+      '';
+    };
+
+    dashboardConfigText = mkOption {
```

Comment:
Why are there json and yaml configs? Do they serve the same purpose? If so, there's no need to allow config for both of them, only the yaml one will be enough.